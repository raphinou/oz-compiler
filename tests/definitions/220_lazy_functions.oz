% Lazy functions. Result variables used in opposite order than unification, and one result unused.
local
   Inc Dec Identity
   R1 R2 R3
in
   fun lazy {Inc N}
      {Show 'Running lazy function Inc with argument:'}
      {Show N}
      N+1
   end
   fun lazy {Dec N}
      {Show 'Running lazy function Dec with argument:'}
      {Show N}
      N-1
   end
   fun lazy {Identity N}
      {Show 'Running lazy function Indentity with argument:'}
      {Show N}
      N
   end

   R1={Inc 100}
   R2={Dec 200}
   R3={Identity 300}
   {Show R2+0}
   {Show R1+0}
end
