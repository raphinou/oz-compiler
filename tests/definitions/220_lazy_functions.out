'Running lazy function Dec with argument:'
200
199
'Running lazy function Inc with argument:'
100
101
